packages/lisk-transactions/src/0_transfer.js

```diff
@@ -12,13 +12,37 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
+import cryptography from 'lisk-cryptography';
 import { BYTESIZES, TRANSFER_FEE } from './constants';
 import {
 	getAddressAndPublicKeyFromRecipientData,
 	wrapTransactionCreator,
+	validateAmount,
 } from './utils';
 
 const createAsset = data => {
+	if (data && data.length > 0) {
+		return { data };
+	}
+	return {};
+};
+
+const validateInputs = ({ amount, recipientId, recipientPublicKey, data }) => {
+	if (!validateAmount(amount)) {
+		throw new Error('Please enter a valid amount!');
+	}
+
+	if (recipientId && recipientPublicKey) {
```

Comment:
@shuse2 I don't think so. This case only covers the situation publicKey and recipientId don't match. if only recipientId is provided then it can go to transaction and when we broadcast get an error message from network if there is any error with it. 

In the schema recipientId was only defined as a string. If there is a unique pattern I can validate but right now I think this is fine ?

Reply:
Reason why `recipientId` is defined in base Transaction is that it is required, but depending on the type of transaction the requirement defers.
For type `0`, it's defined as `format: address`, but in other cases, it is empty string.

But here, I think it should check if the `recipientId` exists.